pkg/plugins/framework/server.go

```diff
@@ -0,0 +1,150 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/lima-vm/lima/pkg/plugins"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+)
+
+// BasePluginServer provides a base implementation for VM driver plugins
+type BasePluginServer struct {
+	plugins.UnimplementedVMDriverServer
+	metadata *plugins.GetMetadataResponse
+	server   *grpc.Server
+}
+
+// NewBasePluginServer creates a new base plugin server
+func NewBasePluginServer(name, version, description string, supportedVMTypes []string) *BasePluginServer {
+	return &BasePluginServer{
+		metadata: &plugins.GetMetadataResponse{
+			Name:             name,
+			Version:          version,
+			Description:      description,
+			SupportedVMTypes: supportedVMTypes,
+		},
+	}
+}
+
+// Start starts the plugin server on the given Unix socket path
+func (s *BasePluginServer) Start(socketPath string) error {
+	// Remove existing socket file if it exists
+	if err := os.RemoveAll(socketPath); err != nil {
+		return fmt.Errorf("failed to remove existing socket: %w", err)
+	}
+
+	// Create Unix domain socket listener
+	lis, err := net.Listen("unix", socketPath)
```

Comment:
You can just use stdio to eliminate the sockets
https://github.com/moby/buildkit/blob/de56a3c5056341667b5bad71f414ece70b50724f/frontend/gateway/grpcclient/client.go#L1240